test/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationTest.java

```diff
@@ -102,9 +112,24 @@ private Subpopulation makeSubpopulation() {
         subpop.setDeleted(true);
         subpop.setAutoSendConsentSuppressed(true);
         subpop.setVersion(3L);
+        subpop.setDataGroupsAssignedWhileConsented(TestConstants.USER_DATA_GROUPS);
+        subpop.setSubstudyIdsAssignedOnConsent(TestConstants.USER_SUBSTUDY_IDS);
         return subpop;
     }
     
+    @Test
+    public void cannotNullifySets() {
+        Subpopulation subpop = new DynamoSubpopulation();
+        // Set some values to verify that null resets these to the empty set
+        subpop.setDataGroupsAssignedWhileConsented(ImmutableSet.of("A"));
+        subpop.setSubstudyIdsAssignedOnConsent(ImmutableSet.of("B"));
+        subpop.setDataGroupsAssignedWhileConsented(null);
+        subpop.setSubstudyIdsAssignedOnConsent(null);
```

Comment:
In between the sets, we should verify that the collections have the expected data in them.

Reply:
Added